xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-tree/xwiki-platform-index-tree-macro/src/main/resources/XWiki/DocumentTreeMacros.xml

```diff
@@ -878,41 +886,65 @@
 #macro (addDocumentContextMenu $contextMenuByNodeType)
   #set ($contextMenuByNodeType.document = {
     'createDocument': {
-      'label': 'New Page',
+      'label': 'New child page',
       'icon': 'fa fa-file-o',
       'action': 'openLink',
       'parameters': {
         'urlProperty': 'createDocumentURL'
       }
     },
-    'openLink': {
+    'edit': {
+      'label': 'Edit',
       'separator_before': true,
-      'label': 'Go to Page',
-      'icon': 'fa fa-external-link'
-    },
-    'refresh': {
-      'label': 'Refresh',
-      'icon': 'fa fa-refresh'
+      'icon': 'fa fa-pencil',
+      'action': 'openLink',
+      'parameters': {
+        'urlProperty': 'editURL'
+      }
     },
-    'cut': {
-      'separator_before': true,
-      'label': 'Cut',
-      'icon': 'fa fa-scissors'
+    'changeTitle': {
```

Comment:
This is a mock operation in the context menu right now, it's not yet implemented.